cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/StatefulSetOperator.java

```diff
@@ -193,15 +195,17 @@ public StatefulSetOperator(Vertx vertx, KubernetesClient client, long operationT
         log.debug("Rolling update of {}/{}: Waiting for pod {} to be deleted", namespace, name, podName);
         Future<Void> podReconcileFuture =
             podOperations.reconcile(namespace, podName, null).compose(ignore -> {
-                Future<Void> del = podOperations.waitFor(namespace, name, pollingIntervalMs, timeoutMs, (ignore1, ignore2) -> {
+                BiPredicate<String, String> predicate = (ignore1, ignore2) -> {
```

Comment:
There's no need for this `BiPredicate`: The lambda parameters are ignored, and you're converting it to a `BooleanSupplier` below anyway, so just declare a `BooleanSupplier`.